src/schemata/number/FloatFromString.ts

```diff
@@ -0,0 +1,163 @@
+/**
+ * Floating point branded newtype from strings. Parameters: min, max are inclusive.
+ *
+ * Note: doesn't technically product lawful encoder / decoders because `toString` is not
+ * symmetric with `Number`.
+ *
+ * Represents string floating point numbers:
+ *
+ * ```math
+ *  { f | f ∈ ℝ, f >= -Number.MAX_VALUE, f <= Number.MAX_VALUE }
+ * ```
+ *
+ * @since 1.0.0
+ */
+import { pipe } from 'fp-ts/function'
+import { SchemaExt, make } from '../../SchemaExt'
+import * as Float from '../../schemables/WithFloat'
+import * as PB from '../../PatternBuilder'
+
+/**
+ * @since 1.0.0
+ * @category Model
+ */
+export type FloatFromStringS = (
+  params?: Float.FloatParams,
+) => SchemaExt<string, Float.Float>
+
+/**
+ * Negative floats with at least one digit before the decimal point.
+ *
+ * E.g. `-1.`; but neither `-.` nor `-`
+ *
+ * @since 1.0.0
+ * @category Pattern
+ */
+const negativeFloatLeft: PB.Pattern = pipe(
+  PB.char('-'),
+  PB.then(pipe(PB.digit, PB.between(1, 308))),
+  PB.then(
+    pipe(PB.char('.'), PB.then(pipe(PB.digit, PB.anyNumber())), PB.subgroup, PB.maybe),
+  ),
+)
+
+/**
+ * Negative floats with at least one digit after the decimal point.
+ *
+ * E.g. `-.1`; but neither `-.` nor `-`
+ *
+ * @since 1.0.0
+ * @category Pattern
+ */
+const negativeFloatRight: PB.Pattern = pipe(
+  PB.char('-'),
+  PB.then(pipe(PB.digit, PB.between(0, 308))),
+  PB.then(
+    pipe(PB.char('.'), PB.then(pipe(PB.digit, PB.atLeastOne())), PB.subgroup, PB.maybe),
+  ),
+)
+
+/**
+ * Positive floats with at least one digit before the decimal point.
+ *
+ * E.g. `1.` but not `.`
+ *
+ * @since 1.0.0
+ * @category Pattern
+ */
+const positiveFloatLeft: PB.Pattern = pipe(
+  PB.digit,
+  PB.between(1, 308),
+  PB.then(
+    pipe(PB.char('.'), PB.then(pipe(PB.digit, PB.anyNumber())), PB.subgroup, PB.maybe),
+  ),
+)
+
+/**
+ * Positive floats with at least one digit after the decimal point.
+ *
+ * E.g. `.1` but not `.`
+ *
+ * @since 1.0.0
+ * @category Pattern
+ */
+const positiveFloatRight: PB.Pattern = pipe(
+  PB.digit,
+  PB.between(0, 308),
+  PB.then(
+    pipe(PB.char('.'), PB.then(pipe(PB.digit, PB.atLeastOne())), PB.subgroup, PB.maybe),
+  ),
+)
+
+/**
+ * @since 1.0.0
+ * @category Pattern
+ */
+const decimalFloat: PB.Pattern = PB.oneOf(
+  negativeFloatLeft,
+  negativeFloatRight,
+  positiveFloatLeft,
+  positiveFloatRight,
+)
+
+/**
+ * @since 1.0.0
+ * @category Pattern
+ */
+const positiveExponential: PB.Pattern = pipe(
+  PB.char('+'),
+  PB.maybe,
+  PB.then(pipe(PB.integerRange(0, 308), PB.subgroup)),
+)
+
+/**
+ * @since 1.0.0
+ * @category Pattern
+ */
+const negativeExponential: PB.Pattern = pipe(
+  PB.char('-'),
+  PB.then(pipe(PB.digit, PB.anyNumber(), PB.subgroup)),
```

Comment:
`atLeastOne`?

Reply:
an invalid test case that (I think) would be marked valid with this: `'1e-'`

Reply:
Good catch!